src/libcollections/vec_deque.rs

```diff
@@ -1232,9 +1232,8 @@ impl<T> VecDeque<T> {
         self.pop_front()
     }
 
-    /// Inserts an element at `index` within the `VecDeque`. Whichever
-    /// end is closer to the insertion point will be moved to make room,
-    /// and all the affected elements will be moved to new positions.
+    /// Inserts an element within the `VecDeque` at index `index`. All elements previously with
+    /// indices greater than or equal to `index` will have their indices incremented by one.
```

Comment:
Perhaps "Inserts an element at position index within the vector, incrementing all indices after `index` by one."

That avoids the problem with somewhat implying the underlying memory operations.